client/wp-admin-scripts/print-shipping-label-banner/shipping-banner/index.js

```diff
@@ -0,0 +1,448 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { Button, ExternalLink } from '@wordpress/components';
+import { compose } from '@wordpress/compose';
+import { recordEvent } from 'lib/tracks';
+import interpolateComponents from 'interpolate-components';
+
+/**
+ * Internal dependencies
+ */
+import '../style.scss';
+import DismissModal from '../dismiss-modal';
+import { getSetting } from '@woocommerce/wc-admin-settings';
+import withSelect from 'wc-api/with-select';
+import SetupNotice, { setupErrorTypes } from '../setup-notice';
+import { withDispatch } from '@wordpress/data';
+import { getWcsAssets, acceptWcsTos } from '../wcs-api';
+
+const wcAdminAssetUrl = getSetting( 'wcAdminAssetUrl', '' );
+
+export class ShippingBanner extends Component {
+	constructor( props ) {
+		super( props );
+
+		const orderId = new URL( window.location.href ).searchParams.get(
+			'post'
+		);
+
+		this.state = {
+			showShippingBanner: true, // TODO: update to get state when closedForever is clicked
+			isDismissModalOpen: false,
+			setupErrorReason: setupErrorTypes.SETUP,
+			orderId: parseInt( orderId, 10 ),
+			wcsAssetsLoaded: false,
+			wcsAssetsLoading: false,
+			wcsSetupError: false,
+			isShippingLabelButtonBusy: false,
+		};
+	}
+
+	componentDidMount() {
+		this.trackImpression();
```

Comment:
Since the `render()` returns `null` if `! showShippingBanner`, we will still be tracking an impression even if the banner isn't rendered. Not sure if that is the intent of this track, but based upon the name... thinking `this.trackImpression()` should only be called if `showShippingBanner` is true?

Reply:
showShippingBanner defaults to true so it'll always show at first unless it's dismissed. We have the PHP code hiding it completely so it shouldn't track the impression unless it's actually shown. However I think it's a good idea to show this check incase we ever change how it works.